### PR TITLE
Refactor the send resource command API

### PR DESCRIPTION
**Pull request by prajeeshprakash**

### CODE REVIEW

File: resource_control_service.c [hunk 1]
```diff
@@ -46,98 +59,215 @@
...
+  coco_std_resource_cmd_t *resourceCmd = (coco_std_resource_cmd_t *)context->params;
+
+  coco_std_free_data(COCO_STD_STRUCT_RESOURCE_CMD, 1, resourceCmd);
+
+  if (-1 == ec_deallocate(context)) {
+    ec_debug_log(LOG_ALERT, "Fatal: Unable to deallocate command context struct, %s, %s",
+                 elear_strerror(elearErrno), ecSuicideString);
+    ec_cleanup_and_exit();
+  }
+
```
Comment (varun-elear) [inline resource_control_service.c]: Don't we need to deallocate `context`?

File: resource_control_service.c [hunk 2]
```diff
@@ -46,98 +59,215 @@
...
-  char *resEui;
-
-  if (-1 == get_gw_node_id_res_eui(resourceHandle, &gwNodeId, &resEui)) {
-    ec_debug_log(LOG_ERR, "Invalid resource handle input", NULL);
+  ec_debug_log(LOG_DEBUG, "Started", NULL);
+  cmd_tx_context_t *context = NULL;
+  net_cphandle_data_t *netData = NULL;
+  uint32_t *seqNum;
+
+  /* Validate the Parameters */
```
Comment (varun-elear) [inline resource_control_service.c]: Use state machine here

File: resource_control_service.c [hunk 3]
```diff
@@ -46,98 +59,215 @@
...
-  if (NULL == commandUmap) {
-    // create table if its not yet created
-    if (NULL == (commandUmap = ec_umap_create(COMMAND_HASH_TABLE_SIZE,
-                                              command_hashing_func, command_key_compare,
-                                              command_free_key_value))) {
-      ec_debug_log(LOG_ALERT, "Issue while creating commandUmap, %s, %s",
+    if (NULL == (deviceMgmtCmdUmap = ec_umap_create(COMMAND_HASH_TABLE_SIZE,
+                                                    command_hashing_func, command_key_compare,
+                                                    command_free_key_value))) {
+      ec_debug_log(LOG_ALERT, "Issue while creating , %s, %s",
```
Comment (varun-elear) [inline resource_control_service.c]: Start with "Fatal: "

File: coco_connection_service.c [hunk 4]
```diff
@@ -336,6 +338,43 @@ static void cp_device_mgmt_cmd_cb(cp_handle_t *cpHandle, void *payload, bool req
...
+                                                       strlen(resrcCmd->resourceEui)))) {
+    ec_debug_log(LOG_ALERT, "Fatal: Unable to duplicate the resourceEui: %s",
+                 resrcCmd->resourceEui);
+    ec_cleanup_and_exit();
+  }
+
+  ec_debug_log(LOG_DEBUG, "Done", NULL);
+  return resrcCmdStatus;
+}
 /**************************************************************************************************
```
Comment (varun-elear) [inline coco_connection_service.c]: Add a blank line

File: resource_control_service.c (hunk 1 above)
Comment (prajeeshprakash) [reply resource_control_service.c]: It is handled

**Pull request merged**